### PR TITLE
DOC: special: Add documentation for `betainc` function

### PR DESCRIPTION
**Pull request by lhilt**
#### Reference issue
None.

#### What does this implement/fix?
Adds a definition, a parameter description, a reference, and some examples of `special.betainc` satisfying some known properties.

#### Additional information
None.

### CODE REVIEW

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -737,22 +737,72 @@ def add_newdoc(name, doc):
...
+        Real-valued such that :math:`0 \leq x \leq 1`,
+        the upper limit of integration
+    out : ndarray, optional
+        Optional output array for the function values
+    
+    Returns
+    -------
+    array-like
+        Value of the incomplete beta function
+
```
Comment (pvanmulbregt) [inline scipy/special/add_newdocs.py]: Suggest adding a "See also:" block here, referring to `beta`, `betaincinv`.

File: scipy/special/add_newdocs.py [hunk 2]
```diff
@@ -737,22 +737,72 @@ def add_newdoc(name, doc):
...
+        Value of the incomplete beta function
+    
     Notes
     -----
-    The incomplete beta is also sometimes defined without the terms
-    in gamma, in which case the above definition is the so-called regularized
-    incomplete beta. Under this definition, you can get the incomplete beta by
-    multiplying the result of the scipy function by beta(a, b).
+    The incomplete beta function is also sometimes defined 
+    without the terms in gamma, in which case the above
```
Comment (pvanmulbregt) [inline scipy/special/add_newdocs.py]: Perhaps "... without the Gamma terms ..."?

File: scipy/special/add_newdocs.py [hunk 3]
```diff
@@ -737,22 +737,72 @@ def add_newdoc(name, doc):
...
+    `1/beta(a, b)`. Also, when :math:`x=1` the integral
+    is equal to `beta(a, b)`. Therefore, :math:`I_x(a, b) = 1`
+    whenever :math:`x = 1`.
+    
+    >>> sc.betainc(0.2, 3.5, 1.0)
+    1.0
+    >>> sc.betainc(9, 0.1, 1.0)
+    1.0
+    >>> sc.betainc(0.0003, 7.4, 1.0)
+    1.0
```
Comment (pvanmulbregt) [inline scipy/special/add_newdocs.py]: There are 3 examples here showing `I_x(a, b) = 1` and one is probably sufficient. Suggest using the space to show other relations, such as to the hypergeometric function. E.g. https://dlmf.nist.gov/8.17#ii
:math:`I_x(a, b) = x^a * F(a, 1-b, a+1, x)/ (a * B(a, b))`

```
>>> a, b, x = 0.2, 3.5, 0.5
>>> x**a * sc.hyp2f1(a, 1-b, a+1, x)/(a*sc.beta(a, b)), sc.betainc(a, b, x)
(0.9896206102333263, 0.9896206102333264)
```

**Review by pvanmulbregt — CHANGES_REQUESTED** [review]
Thanks for adding the documentation.  Just a couple of small changes/additions requested and then it should be good to go.

File: scipy/special/add_newdocs.py (hunk 1 above)
Comment (lhilt) [reply scipy/special/add_newdocs.py]: Done!

File: scipy/special/add_newdocs.py (hunk 2 above)
Comment (lhilt) [reply scipy/special/add_newdocs.py]: Sure. I also made it link to the gamma doc using backticks.

File: scipy/special/add_newdocs.py (hunk 3 above)
Comment (lhilt) [reply scipy/special/add_newdocs.py]: Done.

**Review by pvanmulbregt — APPROVED** [review]

**Pull request merged**